src/sentry/integrations/slack/endpoints/base.py

```diff
@@ -0,0 +1,109 @@
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.base import Endpoint
+from sentry.integrations.slack.message_builder.disconnected import SlackDisconnectedMessageBuilder
+from sentry.integrations.slack.message_builder.help import SlackHelpMessageBuilder
+from sentry.integrations.slack.requests.base import SlackRequestError
+from sentry.integrations.slack.requests.command import SlackCommandRequest
+from sentry.integrations.slack.views.link_identity import build_linking_url
+from sentry.integrations.slack.views.unlink_identity import build_unlinking_url
+
+LINK_USER_MESSAGE = (
+    "<{associate_url}|Link your Slack identity> to your Sentry account to receive notifications. "
+    "You'll also be able to perform actions in Sentry through Slack. "
+)
+UNLINK_USER_MESSAGE = "<{associate_url}|Click here to unlink your identity.>"
+NOT_LINKED_MESSAGE = "You do not have a linked identity to unlink."
+ALREADY_LINKED_MESSAGE = "You are already linked as `{username}`."
+FEATURE_FLAG_MESSAGE = "This feature hasn't been released yet, hang tight."
+
+
+class SlackDMEndpoint(Endpoint):  # type: ignore
+    def post_dispatcher(self, request: Request):
+        """
+        All Slack commands are handled by this endpoint. This block just
+        validates the request and dispatches it to the right handler.
+        """
+        try:
+            slack_request = SlackCommandRequest(request)
```

Comment:
You should pass in a `SlackRequest` instead of always creating a `SlackCommandRequest`.